virtcontainers/kata_agent.go

```diff
@@ -532,6 +532,30 @@ func (k *kataAgent) replaceOCIMountSource(spec *specs.Spec, guestMounts []Mount)
 	return nil
 }
 
+func (k *kataAgent) replaceOCIMountsForStorages(spec *specs.Spec, volumeStorages []*grpc.Storage) error {
+	ociMounts := spec.Mounts
+
+	for i, v := range volumeStorages {
+		for index, m := range ociMounts {
+			if m.Destination != v.MountPoint {
+				continue
+			}
+
+			// Create a temporary location to mount the Storage. Mounting to the correct location
+			// will be handled by the OCI mount structure.
+			filename := fmt.Sprintf("%s-%s", uuid.Generate().String(), filepath.Base(m.Destination))
+			path := filepath.Join(kataGuestSharedDir, filename)
+
+			k.Logger().Debugf("Replacing OCI mount source (%s) with %s", m.Source, path)
+			ociMounts[index].Source = path
+			volumeStorages[i].MountPoint = path
+
+			break
+		}
+	}
+	return nil
```

Comment:
Seems there is no need to have this function returning an error since you return only nil at the end.

Reply:
I am performing some error checking now to return an error if a mount is not found in the list.